openvino_xai/methods/white_box/torch.py

```diff
@@ -86,11 +98,51 @@ def model_forward(self, x: np.ndarray, preprocess: bool = True) -> Mapping:
             output[name] = data.numpy(force=True)
         return output
 
+    def _find_module_by_name(self, model: torch.nn.Module, target_name: str) -> torch.nn.Module:
+        """Search layer by name sub string match."""
+        target_module = None
+        for name, module in model.named_modules():
+            if target_name in name:
+                target_module = module
+        if target_module is None:
+            raise ValueError(f"{target_name} not found in the torch model")
+        return target_module
+
+    def _find_feature_module(self, module: torch.nn.Module) -> torch.nn.Module:
+        """Detect feature module in the model."""
+        # Find the last layer that outputs 4D tensor during temp forward pass
+        self._feature_module = None
+        self._num_modules = 0
+
+        def _detect_hook(module: torch.nn.Module, inputs: Any, output: Any) -> None:
+            if isinstance(output, torch.Tensor):
+                module.index = self._num_modules
+                self._num_modules += 1
+                shape = output.shape
+                if len(shape) == 4 and shape[2] > 1 and shape[3] > 1:
+                    self._feature_module = module
+
+        global_hook_handle = torch.nn.modules.module.register_module_forward_hook(_detect_hook)
+        try:
+            module.forward(torch.zeros((1, 3, 128, 128)))
+        finally:
+            global_hook_handle.remove()
+        if self._feature_module is None:
+            raise RuntimeError("Feature module with 4D output not found in the torch model")
+        if self._feature_module.index / self._num_modules < 0.5:  # Check if ViT-like architectures
+            raise RuntimeError(
+                f"Modules with 4D output end in early-half stages: {100 * self._feature_module.index / self._num_modules}%"
+            )
+
+        return self._feature_module
+
     def _feature_hook(self, module: torch.nn.Module, inputs: Any, output: torch.Tensor) -> torch.Tensor:
+        """Maipulate feature map for saliency map generation."""
```

Comment:
```suggestion
        """Manipulate feature map for saliency map generation."""
```

Reply:
Done in https://github.com/openvinotoolkit/openvino_xai/pull/64/commits/6617ccbf73b700f4cd60646ed49d46673e16060d